### PR TITLE
splinterd add/update/delete node endpoints

### PR DESCRIPTION
**Pull request by ltseeley**
Add the following routes the splinterd REST API:
- POST /nodes for adding nodes to the local registry
- DELETE /nodes/{identity} for deleting nodes from the local registry
- PATCH /nodes/{identity} for updating existing nodes in the local registry

Also moves node registry implementations to libsplinter (will be put behind features(s) in the future), removes the NodeRegistryManager from splinterd, and changes the node registry's create_node to add_node to more accurately reflect the functionality.

### CODE REVIEW

**Comment by ltseeley** [comment]
> This is missing the openapi documentation for the new routes

Added commit to update the OpenAPI spec

File: splinterd/src/daemon.rs [hunk 1]
```diff
@@ -358,6 +358,7 @@ impl SplinterDaemon {
                 routes::get_status(node_id.clone(), service_endpoint.clone())
             }))
             .add_resource(routes::make_fetch_node_resource(node_registry.clone()))
+            .add_resource(routes::make_update_node_resource(node_registry.clone()))
```
Comment (agunde406) [inline splinterd/src/daemon.rs]: Typo in commit message "PATH /nodes/{identity}" -> "PATCH  /nodes/{identity}"

Comment (ltseeley) [reply splinterd/src/daemon.rs]: Fixed

**Comment by ltseeley** [comment]
Added two commits to fix how REST API resources are defined and how the YamlNodeRegistry writes changes to its file.

File: libsplinter/src/rest_api/mod.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@
 //!     fn resources(&self) -> Vec<Resource> {
 //!         let name = self.name.clone();
 //!
-//!         vec![Resource::new(Method::Get, "/index", move |r, p| {
```
Comment (peterschwarz) [inline libsplinter/src/rest_api/mod.rs]: I would suggest that we deprecate the old `new` and make the new way `build`

Comment (ltseeley) [reply libsplinter/src/rest_api/mod.rs]: Done

**Pull request merged**